osu.Game/Updater/UpdateManager.cs

```diff
@@ -62,11 +66,14 @@ protected override void LoadComplete()
             // debug / local compilations will reset to a non-release string.
             // can be useful to check when an install has transitioned between release and otherwise (see OsuConfigManager's migrations).
             config.SetValue(OsuSetting.Version, version);
-        }
 
-        private readonly object updateTaskLock = new object();
+            config.BindWith(OsuSetting.ReleaseStream, releaseStream);
+            releaseStream.BindValueChanged(_ => scheduleUpdateCheck());
```

Comment:
Hmm, this does have a slight edge case which is that if there's a check already running, it will assume that another doesn't need to be run (even thought it probably does because the release stream changed). Might be worth fixing?

See logic in `CheckForUpdateAsync` below when `updateCheckTask` is not null and gets used again as `waitTask`.

Reply:
Check if you agree with the new logic. I had to adjust the inner workings of `UpdateManager` a little bit.